azurelinuxagent/ga/update.py

```diff
@@ -262,9 +262,10 @@ def run(self, debug=False):
             initialize_event_logger_vminfo_common_parameters(protocol)
 
             # Log OS-specific info.
-            os_info_msg = u"Distro info: {0} {1}, osutil class being used: {2}, agent service name: {3}"\
-                .format(DISTRO_NAME, DISTRO_VERSION, type(self.osutil).__name__, self.osutil.service_name)
-
+            os_info_msg = u"Distro info: {0} {1}, osutil class being used: {2}, agent service name: {3}; " \
```

Comment:
can we give some structure to this info? since json seems to cause problems with telemetry maybe a dict? something that is easy to parse in Kusto.

Reply:
The message field isn't a JSON due to the telemetry issue parsing JSON - as you pointed out. I'll try to format the message more deterministically to make sure that it is easy to extract fields in our telemetry language.

eg:
```
Distro: DISTRO_NAME-DISTRO_VERSION; OSUtil: DefaultOSUtil; AgentService: waagent; Python: 3.7.5
```
Which can be easily parsed.

> @larohra - JSON or dict would definitely be ideal 

The telemetry pipeline acts weird when sent JSON, and also when sent dict.__str__() form - eg: the keys are sometimes formatted in Unicode and the parse_json() function in our telemetry fails to parse the dictionary back.